pkg/services/sqlstore/migrator/migrator.go

```diff
@@ -65,6 +103,20 @@ func NewScopedMigrator(engine *xorm.Engine, cfg *setting.Cfg, scope string) *Mig
 	return mg
 }
 
+// Collect implements Prometheus.Collector.
+func (mg *Migrator) Collect(ch chan<- prometheus.Metric) {
+	mg.metrics.migCount.Collect(ch)
+	mg.metrics.migDuration.Collect(ch)
+	mg.metrics.totalMigDuration.Collect(ch)
+}
+
+// Describe implements Prometheus.Collector.
+func (mg *Migrator) Describe(ch chan<- *prometheus.Desc) {
+	mg.metrics.migCount.Describe(ch)
+	mg.metrics.migDuration.Describe(ch)
+	mg.metrics.totalMigDuration.Describe(ch)
+}
```

Comment:
Suggestion: If you use promauto in line 75 to 94 you don't need this.

Reply:
Yeah, I prefer to explicitly register though, so I can handle errors (and in this case, just log a warning and move alone). The main downside with promauto is that it panics on multiple registrations, which are quite possible with this code because a few different code paths create unscoped migrators. They usually shouldn't be in the same process, but...